src/main/java/org/mariadb/jdbc/internal/failover/impl/AuroraListener.java

```diff
@@ -114,6 +149,20 @@ public void reconnectFailedConnection(SearchFilter searchFilter) throws QueryExc
 
         resetOldsBlackListHosts();
 
+        // Initial connection to cluster - retrieve endpoints immediately
+        if (searchFilter.isInitialConnection() && urlParser.getHostAddresses().size() == 1 && getClusterHostAddress() != null) {
+            AuroraProtocol tempProtocol = AuroraProtocol.getNewProtocol(getProxy(), getUrlParser());
+            tempProtocol.setHostAddress(getClusterHostAddress());
+            try {
+                tempProtocol.connect();
+                retrieveAllEndpointsAndSet(tempProtocol);
+            } catch (QueryException e) {
```

Comment:
It is only connecting to the cluster endpoint to retrieve the instances for the very first connection. It is being closed because we want to connect to an instance and not the cluster. This logic is only used on initial connection and is only meant to retrieve the instances.